webapp/src/containers/Rhs/projectDetails/index.tsx

```diff
@@ -103,32 +97,37 @@ const ProjectDetails = memo((projectDetails: ProjectDetails) => {
         },
     });
 
-    // Reset the state when the component is unmounted
-    useEffect(() => {
-        if (!getWebsocketEventState(state).isSubscriptionDeleted) {
-            fetchSubscriptionList();
+    const handleToggleShowAllSubscriptions = (showAllSubscriptions: boolean) => {
+        if (showAllSubscriptions) {
+            setSubscriptionListApiParams({
+                ...subscriptionListApiParams,
+                channel_id: '',
+            });
+        } else {
+            setSubscriptionListApiParams({
+                ...subscriptionListApiParams,
+                channel_id: currentChannelId,
+            });
         }
+    };
```

Comment:
Now we can remove this function. And use `setShowAllSubscriptions` directly in the ToggleSwitch change handler.